### PR TITLE
Add unit tests to cover the migration blocking for bridge network.

### PR DESCRIPTION
**Pull request by petrkotas**
**What this PR does / why we need it**:

Add unit tests to cover the migration blocking for bridge network.

/cc @dhiller @danielBelenky 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-handler/vm_test.go [hunk 1]
```diff
@@ -1151,6 +1151,53 @@ var _ = Describe("VirtualMachineInstance", func() {
...
+						Bridge: &v1.InterfaceBridge{},
+					},
+				},
+			}
+
+			err := controller.checkNetworkInterfacesForMigration(vmi)
+			Expect(err).To(HaveOccurred(), "should block migration for bridging interface")
+		})
+
+		It("should not block migration for bridge interface assigned to pod", func() {
```
Comment (dhiller) [inline pkg/virt-handler/vm_test.go]: s/bridge/masquerade/

**Comment by petrkotas** [comment]
Thanks for the feedback!

@dhiller I have refactored the code using `Context` to make it more readable. I do not like the idea of table. The variables I would  need to pass would still be ugly.

@phoracek added the multus check.

**Comment by phoracek** [comment]
Could you submit the storage changes separately? I don't really see how is that refactoring related to "blocking for bridge".

**Comment by petrkotas** [comment]
Ahh sorry for that, I was trying to save PR.

File: pkg/virt-handler/vm_test.go [hunk 2]
```diff
@@ -1151,6 +1152,82 @@ var _ = Describe("VirtualMachineInstance", func() {
...
+							Masquerade: &v1.InterfaceMasquerade{},
+						},
+					},
+				}
+
+				err := controller.checkNetworkInterfacesForMigration(vmi)
+				Expect(err).ToNot(HaveOccurred(), "should not block migration for masquerade binding method")
+			})
+
+			It("should not block migration for bridge interface assigned to multus network", func() {
```
Comment (phoracek) [inline pkg/virt-handler/vm_test.go]: "should not block migration for bridge binding assigned to a multus network"

File: pkg/virt-handler/vm_test.go [hunk 3]
```diff
@@ -1151,6 +1152,82 @@ var _ = Describe("VirtualMachineInstance", func() {
...
+				vmi := v1.NewMinimalVMI("testvmi")
+				interface_name := "interface_name"
+
+				vmi.Spec.Networks = []v1.Network{
+					v1.Network{
+						Name:          "pod",
+						NetworkSource: v1.NetworkSource{Pod: &v1.PodNetwork{}},
```
Comment (phoracek) [inline pkg/virt-handler/vm_test.go]: you can drop the "pod" network from the test, it is not the subject of this case and there is no `Interface` using it in the following section

File: pkg/virt-handler/vm_test.go [hunk 4]
```diff
@@ -1151,6 +1152,82 @@ var _ = Describe("VirtualMachineInstance", func() {
...
+							Bridge: &v1.InterfaceBridge{},
+						},
+					},
+				}
+
+				err := controller.checkNetworkInterfacesForMigration(vmi)
+				Expect(err).To(HaveOccurred(), "should block migration for bridge binding")
+			})
+
+			It("should not block migration for bridge interface assigned to pod", func() {
```
Comment (phoracek) [inline pkg/virt-handler/vm_test.go]: "should not block migration for *marquerade* binding assigned to the pod network"

File: pkg/virt-handler/vm_test.go [hunk 5]
```diff
@@ -1151,6 +1152,82 @@ var _ = Describe("VirtualMachineInstance", func() {
 			Expect(blockMigrate).To(BeTrue())
 			Expect(err).To(Equal(fmt.Errorf("cannot migrate VMI with non-shared HostDisk")))
 		})
+
+		Context("with network configuration", func() {
+			It("should block migration for bridge interface assigned to pod", func() {
```
Comment (phoracek) [inline pkg/virt-handler/vm_test.go]: "should block migration for bridge binding assigned to the pod network"

**Pull request merged**